content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,519 @@
+
+#### How to calculate percentile.
+There are two conditions that must be satisfied.
+1. Data must be arranged in ascending order.
+2. if the index is not an integer, it must be rounded up.
+```
+np.percentile(df.petal_width, 50)
+```
+![percentile](percentile.png)
+
+ A quartile is another term-related percentile, it can be called a specified percentile, we have 1st quartile as 25% percentile, second quartile as 50%, and 3rd as 75% percentile. You can as well read this [article](https://datascienceupskill.com/statistics/percentiles-and-quartiles) to get a better grip.  
+
+```python
+def petal_cat(df):
+ if df['petal_width'] <= 1.3:
+ return 0
+ elif df['petal_width'] > 1.3:
+ return 1
+ else:
+ return 'indifferent'
+
+df['petal_width_new'] = df.apply(petal_cat, axis=1)  
+```
+The non-machine-friendly format for our *species* class has to be transformed to 0,1 and 2, which is a machine-readable format for the testing to be carried out. Therefore, with [label-encoding](https://www.section.io/engineering-education/introduction-to-scikit-learn-in-python/), a numeric value will be assigned to each of the categories in the class. To learn more about label encoding, [click](https://www.section.io/engineering-education/data-preprocessing-in-r/).
+
+```python
+def species_cat(df):
+ if df["species"] == "Iris-virginica":
+ return 0
+ elif df["species"] == "Iris-versicolor":
+ return 1
+ else:
+ return 2
+
+df["species"] = df.apply(species_cat, axis=1)
+```
+```python
+df.sample(5)
+```
+![dfsample5](dfsample5.png)
+```python
+df_new = df.drop(columns=["sepal_width", "sepal_length", "petal_length", "petal_width"])
+```
+```python
+df_new.head()
+```
+![newhead](df_new.png)
+
+Let's make use of `chic2_contigency` provided for us in `scipy.stats` library.
+
+It has been added to our dependencies earlier in the notebook.
+```python
+stat, p, dof, expected= chi2_contingency(df_new)
+
+print(dof)
+```
+![dof](DOF.png)
+
+We can comfortably get 3 outputs in test statistics, degree of freedom, p-value, and expected values.
+
+[Test statistic](https://online.stat.psu.edu/statprogram/reviews/statistical-concepts/chi-square-tests) is a characterized feature of significance in the Chi-square test that helps in determining how uncommon the result might be provided that null hypothesis is assumed to be true.
```

Comment:
...how uncommon the result might be*, provided* that the null hypothesis is assumed to be true.
Note the comma and the article 'the'.